src/generator/iter.rs

```diff
@@ -8,12 +9,15 @@ use crate::diagnostics::Input;
 use crate::error::{Error, FatalResult, Result};
 use crate::frontend::{Event as FeEvent, EventKind as FeEventKind, Frontend, Include as FeInclude, Graphviz};
 use crate::frontend::range::WithRange;
+use crate::frontend::rustdoc::{Crate, Rustdoc};
 use crate::generator::event::{Event, Tag, Image, Pdf, Svg};
 use crate::generator::Generator;
 use crate::resolve::{Include, ContextType, ResolveSecurity};
 
-pub struct Iter<'a> {
-    frontend: Fuse<Frontend<'a>>,
+type FeEvents<'a> = dyn Iterator<Item=WithRange<FeEvent<'a>>> + 'a;
+
+pub struct MarkdownIter<'a> {
```

Comment:
Is there a reason you're renaming this struct? So far I've followed having the different components in heradoc (frontend, generator, backend), and everyone that can be iterated over has its own Iter struct somewhere. Should it really be called "MarkdownIter"? It doesn't only iterate over markdown, but now also over rustdoc-output.